pyfar/dsp/dsp.py

```diff
@@ -1386,6 +1386,111 @@ def time_shift(signal, shift, unit='samples'):
     return shifted.reshape(signal.cshape)
 
 
+def find_impulse_response_start(ir, N=1):
+    """Find the start of an impulse response.
+    The method relies on the analytic part of the cross-correlation function
+    of the impulse response and it's minimum-phase equivalent, which is zero
+    for the maximum of the correlation function. For sub-sample root finding,
+    the analytic signal is approximated using a polynomial of order ``N``.
+    The algorithm is based on [#]_ with the following modifications:
+
+    1.  Values with negative gradient used for polynolmial fitting are
+        rejected, allowing to use larger part of the signal for fitting.
+    2.  By default a first order polynomial is used, as the slope of the
+        analytic signal should in theory be linear.
+
+
+    Parameters
+    ----------
+    ir : Signal
+        The impulse response
+    N : int, optional
+        The order of the polynom used for root finding, by default 1
+
+    Returns
+    -------
+    numpy.ndarray, float
+        Start samples of the impulse response, can be floating point values
+        in the case of sub-sample values.
+
+    References
+    ----------
+
+    .. [#]  N. S. M. Tamim and F. Ghani, “Hilbert transform of FFT pruned
+            cross correlation function for optimization in time delay
+            estimation,” in Communications (MICC), 2009 IEEE 9th Malaysia
+            International Conference on, 2009, pp. 809-814.
+
+    Examples
+    --------
+    Create a band-limited impulse shifted by 0.5 samples and estimate the
+    starting sample of the impulse and plot.
+
+    .. plot::
+
+        >>> import pyfar as pf
+        >>> import numpy as np
+        >>> from scipy import signal as sgn
+        >>> n_samples = 64
+        >>> sampling_rate = 44100
+        >>> samples = np.arange(n_samples)
+        >>> delay_samples = n_samples // 2 + 1/2
+        >>> sinc = np.sinc(samples - delay_samples)
+        >>> win = sgn.get_window('hanning', n_samples, fftbins=False)
+        >>> ir = pf.Signal(sinc*win, sampling_rate)
+        >>> start_samples = pf.dsp.find_impulse_response_start(ir)
+        >>> ax = pf.plot.time(ir, unit='ms', label='impulse response')
+        >>> ax.axvline(
+        ...     start_samples/sampling_rate*1e3,
+        ...     color='k', linestyle='-.', label='start sample')
+        >>> ax.legend()
+
```

Comment:
We could add additional plots with *real* impulse responses after adding more example data to pyfar.

Reply:
I'd prefer keeping it simple.

Reply:
I've added an example where all peak positions are known. I'd prefer this for illustration over a messy RIR.